packages/contracts-bedrock/src/universal/StandardBridgeExtension.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { StandardBridge } from "./StandardBridge.sol";
+
+abstract contract StandardBridgeExtension is StandardBridge {
+    address public immutable LOCAL_TOKEN;
+    address public immutable REMOTE_TOKEN;
+
+    constructor(
+        address payable _messenger,
+        address payable _otherBridge,
+        address _localToken,
+        address _remoteToken
+    ) StandardBridge(_messenger, _otherBridge) {
+        LOCAL_TOKEN = _localToken;
+        REMOTE_TOKEN = _remoteToken;
+    }
+
+    modifier onlyERC20(address _localToken, address remoteToken) {
+        require(
+            LOCAL_TOKEN == _localToken && REMOTE_TOKEN == remoteToken,
+            "ERC20StandardBridge: can only bridge specified ERC20"
+        );
+        _;
+    }
+
+    /**
+     * Disable ETH Bridging
+     *   - (ideally we could re-route this to the official StandardBridge via the right address from SytemConfig)
+     */
+
+    function _initiateBridgeETH(
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes memory _extraData
+    ) internal pure override {
+        revert("StandardBridgeExtensions: cannot bridge ETH using a standard bridge extension");
+    }
+
+    receive() external override payable {
+        revert("StandardBridgeExtensions: cannot bridge ETH using a standard bridge extension");
+    }
+
+    /**
+     * Bridge only the registered ERC20
+     */
```

Comment:
We have a particular style for the contracts, you should look at them and read the [style guide](https://github.com/ethereum-optimism/optimism/blob/a4f2261b5b12b01009dcd8d6a605eea7fa68b2e1/packages/contracts-bedrock/STYLE_GUIDE.md?plain=1#L2)